fs2-google-pubsub-grpc/src/main/scala/com/permutive/pubsub/producer/grpc/internal/FutureInterop.scala

```diff
@@ -0,0 +1,58 @@
+package com.permutive.pubsub.producer.grpc.internal
+
+import cats.syntax.all._
+import cats.effect.kernel.{Async, Sync}
+import com.google.api.core.{ApiFuture, ApiFutureCallback, ApiFutures}
+import com.google.common.util.concurrent.MoreExecutors
+
+private[internal] object FutureInterop {
+  def fFromFuture[F[_]: Async, A](future: F[ApiFuture[A]]): F[A] =
+    future.flatMap { futA =>
+      Async[F]
+        .async[A] { cb =>
+          Async[F].delay {
+            val futureApi = futA
+            addCallback(futureApi)(cb)
+            Option(
```

Comment:
Are you using an `Option` instead of a `Some` here to help the compiler?

Reply:
Yeah I had some issues with the compiler being confused, hence I change it to `Option`. I think it's it do with variance?

Reply:
Yeah, thats why cats has extra option syntax for some: `Option.some`

Reply:
Actually seems like it compiles fine now 🤷  Maybe was something else. Changed in 3222e09.